### PR TITLE
More syslog API

### PR DESCRIPTION
**Pull request by lvh**

### CODE REVIEW

File: src/unclogged/core.clj [hunk 1]
```diff
@@ -51,6 +54,79 @@
...
+  [conn-details]
+  (let [{:keys [host port message-format transport]
+         :or {:transport :tls}} conn-details
+        syslog (make-syslog transport)]
+    (.setSyslogServerHostname syslog host)
+    (when port
+      (.setSyslogServerPort syslog port))
+    (when message-format
+      (.setMessageFormat syslog (parse-message-format message-format)))
+    (when (#{:tls :ssl} transport)
```
Comment (derwolfe) [inline src/unclogged/core.clj:88]: Should this also include API to configure where to find the public key when using tls?

Comment (lvh) [inline src/unclogged/core.clj:88]: It should, but this isn't current supported in the upstream library first. It just kinda gets a default Java TLS socket.

File: src/unclogged/core.clj [hunk 2]
```diff
@@ -51,6 +54,79 @@
...
+
+  Values in message maps override values in the defaults; values in
+  the defaults override unclogged's default-defaults. This prevents
+  NullPointerExceptions when you forget to provide a facility or
+  severity, for example.
+
+  Returns the source, annotated with :unclogged/syslog in its
+  metadata. This is only provided for inspection; mutating that object
+  is not guaranteed to have desired effects."
+  [source conn-details defaults]
```
Comment (derwolfe) [inline src/unclogged/core.clj]: Are there any differences between `conn-details` here and `conn-opts` at line 110?

Comment (derwolfe) [inline src/unclogged/core.clj]: If not, it seems like it would be clearer were they to use the same name.

Comment (lvh) [inline src/unclogged/core.clj]: You mean at line 125? No, that's a stupid inconsistency. Which one do you prefer?

File: src/unclogged/core.clj [hunk 3]
```diff
@@ -51,6 +54,79 @@
...
+
+(defn syslog-sink
+  "Returns a Manifold sink (stream) where you can dump information
+  you'd like to send to syslog. Takes a connection map and some
+  message defaults.
+
+  If you already have a manifold stream, see ->syslog!."
+  [conn-opts defaults]
+  (let [stream (s/stream)]
+    (->syslog! stream)
```
Comment (lvh) [inline src/unclogged/core.clj]: When you do not test a thing, it will have a bug.

File: src/unclogged/core.clj (hunk 2 above)
Comment (derwolfe) [inline src/unclogged/core.clj]: Yes - 125. I'd go with `conn-opts`.

File: src/unclogged/core.clj (hunk 3 above)
Comment (derwolfe) [inline src/unclogged/core.clj]: Are you pointing out that this did not take `conn-opts`?

Comment (lvh) [inline src/unclogged/core.clj]: yup.

**Pull request merged**